app/pages/lab/actions/workflow.js

```diff
@@ -3,12 +3,23 @@ var apiClient = require('panoptes-client/lib/api-client');
 var workflows = apiClient.type('workflows');
 
 var workflowActions = {
-  createWorkflowForProject: function(projectID, workflowData) {
+  createWorkflowForProject: function(project, workflowData) {
     var allWorkflowData = Object.assign({
-      links: {project: projectID}
+      links: {project: project.id}
     }, workflowData);
 
     return workflows.create(allWorkflowData).save();
+  },
+  copyWorkflowForProject: async function(project, workflowData) {
+    await project.addLink('workflows', [workflowData.id]);
```

Comment:
i think the API should ensure we don't lose the workflow links if folks 'forget' to wrap the params in an array...that's terrible on our behalf and i can't see a usecase for overwriting the project's worklfow links. 

That said it runs the risk of breaking some clients expectations on API behaviours, https://www.hyrumslaw.com/

That said - I will change the API to ensure we never delete those links.

Reply:
The API client has a very old TODO saying ‘should this always be an array?’ That can be enforced in JavaScript too. 
https://github.com/zooniverse/json-api-client/blob/f4cdb5498df385b38b3bd51a11ed80d44f14ea31/src/resource.coffee#L187

So old that it’s written in CoffeeScript.

Reply:
there are times when the link can be a singular resource, i.e. a project has one avatar etc so the array form doesn't make sense there. I think pushing the responsibility API side is a better approach than forcing the clients to know what kind of relation they are updating and which form is correct. 

> So old that it’s written in CoffeeScript.
It's a little worrying that a lot of our front end code relies on this client code, even the FEM code includes it some resources.

Reply:
Zoo Notes uses the new API client. I think that's a question of getting people out of the ingrained habit of using the old client. Or finding out why they're avoiding the new client and fixing it.